selectel/resource_selectel_iam_saml_federation_certificate_v1.go

```diff
@@ -0,0 +1,164 @@
+				Computed:    true,
+				Description: "Certificate lifetime right bound.",
+			},
+			"fingerprint": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Fingerprint.",
+			},
+		},
+	}
+}
+
+func resourceIAMSAMLFederationCertificateV1Create(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	iamClient, diagErr := getIAMClient(meta)
+	if diagErr != nil {
+		return diagErr
+	}
+
+	log.Print(msgCreate(objectSAMLFederationCertificate, d.Id()))
+	federation, err := iamClient.SAMLFederations.Certificates.Create(ctx, d.Get("federation_id").(string), certificates.CreateRequest{
+		Name:        d.Get("name").(string),
+		Description: d.Get("description").(string),
+		Data:        d.Get("data").(string),
+	})
+	if err != nil {
+		return diag.FromErr(errCreatingObject(objectSAMLFederationCertificate, err))
+	}
+
+	d.SetId(federation.ID)
+
+	return resourceIAMSAMLFederationCertificateV1Read(ctx, d, meta)
+}
+
+func resourceIAMSAMLFederationCertificateV1Read(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	iamClient, diagErr := getIAMClient(meta)
+	if diagErr != nil {
+		return diagErr
+	}
+
+	log.Print(msgGet(objectSAMLFederationCertificate, d.Id()))
+	certificate, err := iamClient.SAMLFederations.Certificates.Get(ctx, d.Get("federation_id").(string), d.Id())
+	if err != nil {
+		return diag.FromErr(errGettingObject(objectSAMLFederationCertificate, d.Id(), err))
+	}
+
+	d.Set("account_id", certificate.AccountID)
+	d.Set("name", certificate.Name)
+	d.Set("description", certificate.Description)
+	d.Set("not_before", certificate.NotBefore)
+	d.Set("not_after", certificate.NotAfter)
+	d.Set("fingerprint", certificate.Fingerprint)
+	d.Set("data", certificate.Data)
+
+	return nil
+}
+
+func resourceIAMSAMLFederationCertificateV1Update(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	iamClient, diagErr := getIAMClient(meta)
+	if diagErr != nil {
+		return diagErr
+	}
+
+	desc := d.Get("description").(string)
+
+	opts := certificates.UpdateRequest{
+		Name:        d.Get("name").(string),
+		Description: &desc,
+	}
+
+	log.Print(msgUpdate(objectSAMLFederationCertificate, d.Id(), opts))
+	_, err := iamClient.SAMLFederations.Certificates.Update(ctx, d.Get("federation_id").(string), d.Id(), opts)
+	if err != nil {
+		return diag.FromErr(errUpdatingObject(objectSAMLFederationCertificate, d.Id(), err))
+	}
+
+	return resourceIAMSAMLFederationCertificateV1Read(ctx, d, meta)
+}
+
+func resourceIAMSAMLFederationCertificateV1Delete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	iamClient, diagErr := getIAMClient(meta)
+	if diagErr != nil {
+		return diagErr
+	}
+
+	log.Print(msgDelete(objectSAMLFederationCertificate, d.Id()))
+	err := iamClient.SAMLFederations.Certificates.Delete(ctx, d.Get("federation_id").(string), d.Id())
+	if err != nil && !errors.Is(err, iamerrors.ErrFederationNotFound) {
```

Comment:
Should we handle “certificate not found” error here?

Reply:
Unfortunately, there is no `ErrFederationCertificateNotFound`, so I decided to print any error

Reply:
I think we should add this error to iam-go, to handle such situations in a consistent way.

Reply:
Done!